inc/zoo/swar/SWAR.h

```diff
@@ -77,6 +77,26 @@ struct SWAR {
 
     constexpr T value() const noexcept { return m_v; }
 
+    // is this the right name?
+    constexpr static T maxLaneValue = (T{1} << NBits) - 1;
```

Comment:
Never shift by `NBits`, if the SWAR is "degenerated", in the sense that it contains only one lane, then you would be shifting all bits, the result is undefined.  This gave Scott and I a problem preparing for CPPCon 2022, it turns out that even between processors of the same brand, the result of shifting by the whole bit count is different, sometimes it is zero, some others does nothing.
Here's what you want:

```suggestion
    constexpr static MaxUnsignedLaneValue = {}() { // a lambda to give names to the intermediate steps
        T
            numberOneInTheLeastSignificantLane{1},
            turnedToMSB = numberOneInTheLeastSignificantLane << (NBits - 1),
            leastSignificantLaneFlipped = turnedToMSB - 1,
            rv = turnedToMSB | leastSignificantLaneFlipped;
        return rv;
    }();        
```
I did not make the rules.  This is exceedingly strange, that something as simple as this requires something so complicated.
Alternatively you can `~(((~T{0}) << (NBits - 1)) << 1)` or whatever else you want, but what you wrote is wrong, and in practice would lead you to get corrupt code silently.